server/src/sim.rs

```diff
@@ -257,16 +322,28 @@ impl Sim {
             let id = *self.world.get::<&EntityId>(entity).unwrap();
             spawns.push((id, dump_entity(&self.world, entity)));
         }
-        if !self.graph.fresh().is_empty() {
+
+        let mut modified_chunks = vec![];
+        for fresh_node in fresh_nodes.iter().copied() {
+            for vertex in Vertex::iter() {
+                let chunk = ChunkId::new(fresh_node, vertex);
+                if let Some(voxel_data) = self.pending_modified_chunks.remove(&chunk) {
+                    modified_chunks.push((chunk, voxel_data.to_serializable(self.cfg.chunk_size)));
+                    self.modified_chunks.insert(chunk);
+                    self.graph.populate_chunk(chunk, voxel_data, true)
+                }
+            }
+        }
```

Comment:
There's too many different things named "modified_chunks" or a variation of it here. We need more distinctive names and clear documentation of what each one represents.

Reply:
Good point. I definitely agree, especially since both client's `sim` and server's `sim` have a field called `pending_modified_chunks`, but one uses a `Vec<BlockUpdate>` while the other uses a `VoxelData`.

It's also weird that this is the name for both lists of chunks that were modified and maps that say what the modifications were, since those two things are conceptually quite different.

Reply:
I think I found reasonable names for all these variables.
- The two versions of `pending_modified_chunks` are now `preloaded_block_updates` and `preloaded_voxel_data`
- The three versions of `modified_chunks` are now `voxel_data`, `modified_chunks`, and `fresh_voxel_data`